cloudwatch.tf

```diff
@@ -0,0 +1,65 @@
+resource "aws_cloudwatch_log_group" "main" {
+  name              = "/ecs/${var.environment}/github-runner"
+  retention_in_days = var.cloudwatch_log_retention
+
+  kms_key_id = aws_kms_key.log_enc_key.arn
+
+  tags = {
+    Name        = "github-runner"
+    Environment = var.environment
+    Automation  = "Terraform"
+  }
+
+  lifecycle {
+    prevent_destroy = true
+  }
+
+}
+
+resource "aws_kms_key" "log_enc_key" {
+  description         = "KMS key for encrypting logs"
+  enable_key_rotation = true
+  policy              = data.aws_iam_policy_document.cloudwatch_logs_allow_kms.json
+
+  tags = {
+    Automation = "Terraform"
+  }
+}
+
+data "aws_iam_policy_document" "cloudwatch_logs_allow_kms" {
+  statement {
+    sid    = "Enable IAM User Permissions"
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root",
+      ]
+    }
+
+    actions = [
+      "kms:*",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "Allow logs KMS access"
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["logs.${data.aws_region.current.name}.amazonaws.com"]
+    }
+
+    actions = [
+      "kms:Encrypt*",
+      "kms:Decrypt*",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:Describe*"
+    ]
+    resources = ["*"]
```

Comment:
Looks like you can scope down access to the KMS log group defined in this module via
```
            "Condition": {
                "ArnEquals": {
                    "kms:EncryptionContext:aws:logs:arn": "arn:aws:logs:region:account-id:log-group:log-group-name"
                }
```

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html

Reply:
Ah, nice! I think I've got it translated here now in my latest commit - does that look right?

Reply:
I think for line 70, you can just look up the arn
```
values = [
        aws_cloudwatch_log_group.main.arn
      ]
```